homeassistant/components/ai_task/task.py

```diff
@@ -104,20 +89,58 @@ async def async_generate_data(
                 )
             )
 
-    with async_get_chat_session(hass) as session:
-        if created_files:
+    if created_files:
```

Comment:
let's make this a guard clause: `if not created_files: return resolved_attachments`